Userland/Libraries/LibCore/Process.cpp

```diff
@@ -163,6 +167,35 @@ ErrorOr<bool> Process::is_being_debugged()
         return (tracer_pid != 0UL);
     }
     return false;
+#elif defined(AK_OS_GNU_HURD)
+    process_t proc = getproc();
+    if (!MACH_PORT_VALID(proc))
+        return Error::from_syscall("getproc"sv, -errno);
+
+    int flags = PI_FETCH_TASKINFO;
+    struct procinfo pi_buffer;
```

Comment:
can we zero-init this guy? `struct procinfo pi_buffer = {}`

Reply:
We could , but why? It's going to be written over (or not) by the `proc_getprocinfo()` call.

Reply:
I would rather be explicit than rely on the kernel/glibc to overwrite all the padding bits with 0, or write all of the fields that we end up accessing later. If the function is actually a macro, the compiler should be able to eliminate any dead stores. And it's not like this is a perf-critical path.

Reply:
(It's not a macro), but sure, let's initialize it then. This is certainly not performance critical compared to the actual RPC 🙂